pages/Documents/ConversationalAI/Third-PartyBots/AmazonLex/advanced-features.md

```diff
@@ -317,3 +317,78 @@ Once all of above steps has been configured and updated bot has been published t
+
+### Receiving Last consumer message (Messaging Only)
+
+Third-Party Bots will provide the last message sent by the consumer as part of the welcome event when an ongoing conversation gets transferred to a new Agent or skill. This will provide a way to run custom logic and respond back with a different response than the normal welcome intent.
+
+An example use case of the last consumer message sent by Third-Party Bots is described below. The example will show how to set up and access the WelcomeEvent response in Amazon Lex. We will use Amazon Lex capability of providing fulfillment via Amazon Lambda.
+
+#### Create Amazon Lambda Function
+
+ Last consumer message is part of context information sent by Third-Party Bots in the request. For accessing the WelcomeEvent body we will need to create an Amazon Lambda function that should be capable of parsing the additional message context sent by Third-Party Bots. The minimal code example below shows how to check if there is a `LastConsumerMessage` present in the message context , then send back text response containing the last consumer message. Please note that response should follow the Lex response schema and for more information on how to create Lambda function for Amazon Lex you can also follow [the official documentation](https://docs.aws.amazon.com/lex/latest/dg/gs2-prepare.html)
+
+
+```javascript
+'use strict';
+
+function close(sessionAttributes, fulfillmentState, message) {
+  return {
+    sessionAttributes,
+    dialogAction: {
+      type: 'Close',
+      fulfillmentState,
+      message,
+    },
+  };
+}
+
+// --------------- Events -----------------------
+
+function dispatch(intentRequest, callback) {
+  const sessionAttributes = intentRequest.sessionAttributes;
+  const requestAttributes = intentRequest.requestAttributes;
+  let response;
+
+  if (requestAttributes && requestAttributes['BC-LP-CONTEXT']) {
+    const parsedLpContext = JSON.parse(requestAttributes['BC-LP-CONTEXT']);
+    if (parsedLpContext.lastConsumerMessage) {
+      response = `I received lastConsumerMessage: ${parsedLpContext.lastConsumerMessage}`;
+    }
+    else {
+      response = 'Unable to find any lastConsumerMessage data';
+    }
+  }
+  else {
+    response = 'Unable to find any LP Context Information';
+  }
+
+  callback(close(sessionAttributes, 'Fulfilled', { 'contentType': 'PlainText', 'content': response }));
+}
+
+// --------------- Main handler -----------------------
+
+// Route the incoming request based on intent.
+// The JSON body of the request is provided in the event slot.
+exports.handler = (event, context, callback) => {
+  try {
+    dispatch(event, (response) => {
+      callback(null, response);
+    });
+  }
+  catch (err) {
+    callback(err);
+  }
+};
+```
+
+
+#### Create WelcomeEvent Intent and Link Amazon Lambda
+
+Ensure you have an ‘entry point’ intent that utilises the default ‘WELCOME’ event, if not you can create new intent triggred by `WELCOME-INTENT` utterances.
```

Comment:
```
Ensure you have an ‘entry point’ intent that utilizes the default ‘WELCOME’ event, 
if not you can create new intent triggered by `WELCOME-INTENT` utterances.
After the Sample utterances are added move to the section of the fulfillment 
and choose the AWS Lambda function. Our deployed lambda function should 
populate the list of selections. Select the Lambda function which in our example 
case is WelcomeEventCloudFunction. This can be seen in Figure 3.9 below
```